### ISSUE TITLE
Unable to locate build via Github actions API

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [codecov/feedback#180]
**Describe the bug**
I'm using the Codecov uploader Github Actions in my CI and we started noticing an upload error earlier today.

https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667

```
Run codecov/codecov-action@v3
  with:
  env:
    INPUT_RUN_POST: true
    CONDA_PKGS_DIR: /home/runner/conda_pkgs_dir
==> linux OS detected
https://uploader.codecov.io/latest/linux/codecov.SHA[2](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:2)56SUM
==> SHASUM file signed by key id 806bb28aed779869
==> Uploader SHASUM verified (b9282b8b4[3](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:3)eef83f7226[4](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:4)6d8992c4dd36[5](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:5)[6](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:7)3046afe0806[7](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:8)221[8](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:9)4e7e[9](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:10)923a6d7b  codecov)
==> Running version latest
==> Running version v0.7.1
/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov -n  -Q github-action-3.1.4 -C ded5bded4e9db819543e7b5bf2d69a50009148ef
[2023-12-[11](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:12)T17:14:11.531Z] ['info'] 
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/

  Codecov report uploader 0.7.1
[2023-[12](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:13)-11T17:[14](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:15):11.539Z] ['info'] => Project root located at: /home/runner/work/FESTIM/FESTIM
[2023-12-11T[17](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:18):14:11.540Z] ['info'] -> No token specified or token is empty
[2023-12-11T17:14:11.549Z] ['info'] Searching for coverage files...
[2023-12-11T17:14:11.600Z] ['info'] Warning: Some files located via search were excluded from upload.
[2023-12-11T17:14:11.600Z] ['info'] If Codecov did not locate your files, please review https://docs.codecov.com/docs/supported-report-formats
[2023-12-11T17:14:11.600Z] ['info'] => Found 1 possible coverage files:
  coverage.xml
[2023-12-11T17:14:11.600Z] ['info'] Processing /home/runner/work/FESTIM/FESTIM/coverage.xml...
[2023-12-11T17:14:11.603Z] ['info'] Detected GitHub Actions as the CI provider.
[2023-12-11T17:14:11.887Z] ['info'] Pinging Codecov: https://codecov.io/upload/v4?package=github-action-3.1.4-uploader-0.7.1&token=*******&branch=update-codecov-upload&build=7170749622&build_url=https%3A%2F%2Fgithub.com%2Ffestim-dev%2FFESTIM%2Factions%2Fruns%2F7170749622%2Fjob%2F[19](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:20)524611126&commit=ded5bded4e9db819543e7b5bf2d69a50009148ef&job=CI&pr=667&service=github-actions&slug=festim-dev%2FFESTIM&name=&tag=&flags=&parent=
[[20](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:21)[23](https://github.com/festim-dev/FESTIM/actions/runs/7170749622/job/19524611126?pr=667#step:7:24)-12-11T17:14:12.079Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Environment (please complete the following information):**
Github Actions (Linux)

**Additional context**
I've tried:
- Re-running the workflow
- Updating the CODECOV secret in the repo
- update the upload step using the marketplace Codecov action

**Comment by RemDelaporteMathurin** [comment]
> It looks like it's not locating the token. How are you passing the repo secret to the action?
> 
> It should be via
> 
> ```
>   with:
>     token: ${{ secrets.CODECOV_TOKEN }}
> ```

Hi! Yes, this is how I'm passing it:

```yaml
    - name: Upload coverage to Codecov
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
```

Plus, it used to work earlier today without any change to the action itself.

**Comment by thomasrockhu-codecov** [comment]
@RemDelaporteMathurin, this is strange, let's see, can you try

1. Double-checking that `CODECOV_TOKEN` is set as a repository secret?
2. Use
```
   - name: Upload coverage to Codecov
      uses: codecov/codecov-action@v3
      env:
        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

**Comment by RemDelaporteMathurin** [comment]
This morning it started working again without any change from our side overnight.

🟢  [Working workflow this morning](https://github.com/festim-dev/FESTIM/actions/runs/7178919862/job/19549230132).
🔴 [Failing workflow yesterday](https://github.com/festim-dev/FESTIM/actions/runs/7169382301/job/19521897845).

The [workflow file](https://github.com/festim-dev/FESTIM/actions/runs/7178919862/workflow) hasn't changed whatsoever



> 1. Double-checking that `CODECOV_TOKEN` is set as a repository secret?

Yes





> 2\. Use
> 
> ```
>     - name: Upload coverage to Codecov
>       uses: codecov/codecov-action@v3
>       env:
>         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
> ```

The upload step is:

```
    - name: Upload to codecov
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
        files: ./coverage.xml
```
As indicated in [the documentation of the action](https://github.com/marketplace/actions/codecov#usage).

**Comment by RemDelaporteMathurin** [comment]
Overall, even before yesterday (and on other repos) we often have intermittent upload issues with codecov that require either to re-run jobs hoping that the upload will work or (like now) wait a few hours/days and try again hoping that things will work later

**Comment by drazisil-codecov** [comment]
Hi @RemDelaporteMathurin ,

The reason it works sometimes is because when Codecov can't find the upload token for a public repo, we use tokenless, which allows us to poll GitHub to check that this is a valid and current CI run. The reason it fails sometimes, is that we are using our "personal" API token for Codecov, and we hit GitHub's rate limits.

In order to avoid this, we need to figure out why the action isn't seeing your token, so we can use that instead.

You can see that, even on the passing build from this morning, we are still not seeing your token. https://github.com/festim-dev/FESTIM/actions/runs/7178919862/job/19549230132#step:7:26

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov thanks for your reply. This is a screenshot of the secrets tab of the repository. the CODECOV_TOKEN is there

![image](https://github.com/codecov/feedback/assets/40028739/89853db0-ba0e-4990-bdb0-35fd478b5de1)

**Comment by drazisil-codecov** [comment]
Hi @RemDelaporteMathurin ,

I didn’t mean to imply I didn't believe you. 😅 

Can you add verbose to the action? Maybe that will give us more clues as to why it's not being seen. 🤔

**Comment by RemDelaporteMathurin** [comment]
> I didn’t mean to imply I didn't believe you. 😅

Oh and I didn't mean to imply you implied this 🙃



> Can you add verbose to the action? Maybe that will give us more clues as to why it's not being seen

I'm on it.

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov please find the results of the verbose here
https://github.com/festim-dev/FESTIM/actions/runs/7182795041/job/19560097608?pr=668

**Comment by drazisil-codecov** [comment]
I'm stumped. 😖 

I know you tried it, but do you mind trying
```
      env:
        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
        ```
        Again, while we still have verbose enabled? Turns out we've swapped all our examples to 4-beta, and I don't have a. personal example whereI'm using the action. In the meanwhile, I'll fire up a repo and make sure this actually works.

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov 

Sure! Thanks for trying to solve this! 👍 
Here you go: https://github.com/festim-dev/FESTIM/actions/runs/7183704934/job/19563071125?pr=668

**Comment by drazisil-codecov** [comment]
Well, it works HERE

https://github.com/drazisil-codecov/python-standard/actions/runs/7183742467/job/19563187036#step:6:52

https://github.com/drazisil-codecov/python-standard/actions/runs/7183742467/job/19563187036#step:6:90

Let me take a look at yours...

**Comment by drazisil-codecov** [comment]
I... have zero clue why yours isn't reading the token.

It looks identical to mine. Um. I would refer you to GitHub support, but they would probably just send you back to me.

@thomasrockhu-codecov do you have any ideas?

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov is there some kind of way to not grant Codecov access to the repo's secrets?

**Comment by drazisil-codecov** [comment]
@RemDelaporteMathurin 

Not that I am aware of. I looked for settings to that effect, but I didn't see anything . :-/

**Comment by drazisil-codecov** [comment]
@RemDelaporteMathurin 

If you set it manually to gibberish in the yaml, can we confirm that that part at least is working, in that the action would see it if it was at least getting to the yml?

**Comment by RemDelaporteMathurin** [comment]
Sure, here we are:

https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7184336930/job/19565167932

This is the current yaml:

```yaml
codecov:
  token: "blablablablablabla"
ignore:
  - "festim/_version.py"  # ignore version file
```

**Comment by drazisil-codecov** [comment]
Well, that worked. So the issue is with GitHub not passing the Secret, not the action. I guess that's....progress? 🤔

**Comment by drazisil-codecov** [comment]
Oh. No, I mean in the workflow yaml, sorry.

**Comment by RemDelaporteMathurin** [comment]
Oh sorry! I moved it to the workflow YAML: https://github.com/festim-dev/FESTIM/actions/runs/7184562840/job/19565899192?pr=668

```yaml
    - name: Upload to codecov
      uses: codecov/codecov-action@v3
      env:
        CODECOV_TOKEN: "blablablabla"
      with:
        files: ./coverage.xml
        verbose: true
```

**Comment by drazisil-codecov** [comment]
That worked 🥳 

https://github.com/festim-dev/FESTIM/actions/runs/7184562840/job/19565899192?pr=668#step:7:60

So something’s up with passing the secret across to the action. 🤔

**Comment by drazisil-codecov** [comment]
Seems like potentially a random issue with no fix https://github.com/orgs/community/discussions/25701

What happens if you try setting the secret as a top-level env for the workflow, instead of at the step level?

**Comment by RemDelaporteMathurin** [comment]
Happening there: https://github.com/festim-dev/FESTIM/actions/runs/7184707778/job/19566395040?pr=668

**Comment by drazisil-codecov** [comment]
Hrm. Does it work as an environmental secret? I think I saw you using those. Does your other action secret still work?

**Comment by RemDelaporteMathurin** [comment]
Right, didn't work.

Maybe something's wrong with how I defined the secret in the first place? Can we do a little sanity check:
When copying the secret from the codecov website, it's in the `CODECOV_TOKEN=.......` format (see below).
![image](https://github.com/codecov/feedback/assets/40028739/7f561d1f-b4bb-4d46-bec6-0f8ed66b1c1d)

Should I put everything (including `CODECOV_TOKEN=`) in the secret value?

**Comment by drazisil-codecov** [comment]
Nope. Just the part to the right of the equals. @codecovdesign for product feedback to make that clearer.

**Comment by drazisil-codecov** [comment]
... I would call that a bug that it didn't warn you, but since I don't know how its passed to the action in that case (if at all), I don't know if the action has any way to know. 🤔

**Comment by RemDelaporteMathurin** [comment]
> Nope. Just the part to the right of the equals

Right that's what I thought but you know... this is so obscure...

I went back to having the secret as the job level and tried updating the secret value (without the `CODECOV_TOKEN=` of course)

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov and it didn't work...
https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7184905565/job/19567033940#step:7:61

**Comment by drazisil-codecov** [comment]
Drat! I thought you had solved it. :-/

I'm still poking, but without being able to reproduce...

If you fork another repo and try, does that one work?

**Comment by drazisil-codecov** [comment]
Does your pypi api token. still work?

I'd love to be able to definitively say it's not a Codecov issue...somehow.

**Comment by RemDelaporteMathurin** [comment]
> If you fork another repo and try, does that one work?

I am trying here https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7184975439
I have codecov setup for this fork and the fork has a secret too

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov it worked on the fork!!
https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7184975439/job/19567340533#step:8:60

**Comment by drazisil-codecov** [comment]
That worked! https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7184975439/job/19567340533#step:8:60

**Comment by drazisil-codecov** [comment]
Well, at this point I can say either user error, or GitHub. I hope it's the second one. Please let me know if you figure it out. 🙏

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov something worth mentionning.

RemDelaporteMathurin/FESTIM used to be the only repo. We transfered the ownership to festim-dev/FESTIM recently.

Is it possible that this messed things up?

**Comment by drazisil-codecov** [comment]
Maybe poke that issue about it being random. It seems to match that pattern.

**Comment by drazisil-codecov** [comment]
> Is it possible that this messed things up?

It shouldn't. You can try rolling the upload token if you want, but at the point it's failing, Codecov doesn't even know you exist yet.

**Comment by RemDelaporteMathurin** [comment]
Right so likely a GitHub issue then. Do you know where I can open an issue?

**Comment by drazisil-codecov** [comment]
Either add on to https://github.com/orgs/community/discussions/25701, or start a new one and mention it, is what I would do.

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov i defined the secret at the job level and it WORKED! 🎉 
It is definitely a problem of Github not passing the secret correctly.

https://github.com/RemDelaporteMathurin/FESTIM/actions/runs/7185146381/job/19567781471#step:7:58

Thank you for helping out with this!! 👍

**Comment by RemDelaporteMathurin** [comment]
Update: well... it still seems to be randomely failing to pass the secret
https://github.com/festim-dev/FESTIM/actions/runs/7185279996/job/19568226731?pr=668#step:7:27

**Comment by RemDelaporteMathurin** [comment]
@drazisil-codecov it seems my brain just turned to mush. When I thought it worked, I was looking at the forked repo not the org one.

Turns out it's rather simple and not a bug at all:

Organisation secrets are not available from external PRs (https://github.com/codecov/codecov-action/issues/29).

I don't think there a way to solve this issue and we'll just have to rely on the token-less upload (unless you are aware of another solution).

I tested this workflow from a branch of the organisation repo (not the fork) and the token is [found correctly](https://github.com/festim-dev/FESTIM/actions/runs/7187084497/job/19573886435#step:9:28).

Thanks again for looking into this with me!
